### PR TITLE
add ide ignore files

### PR DESCRIPTION
**Pull request by zouyee**
## Proposed Changes

* add ide ignore files

**Release Note**

```NONE

```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,6 +15,7 @@
 
 # Fortran module files
 *.mod
+*.sum
```
Comment (vdemeester) [inline .gitignore]: Why `*.sum` ?
If we ever switch to `Go` modules `*.sum` and `*.mod` we will need to add them back

Comment (zouyee) [reply .gitignore]: Currently we have not modified the vendor tool to `go mod`, so we need to exclude `go.sum` and `go.mod`, also, when somebody using IDE, the above two files may be automatically generated.

**Comment by ImJasonH** [comment]
Closing this as stale, please reopen if you think this is still something we should do.

**Pull request closed without merging**